azure-pipelines.yml

```diff
@@ -1,162 +1,580 @@
+                  cacheHitVar: CYPRESS_CACHE_RESTORED
+                displayName: 'Cache Cypress binary'
+              - task: npmAuthenticate@0
+                inputs:
+                  workingFile: '$(AppHostingSdkProjectDirectory)/.npmrc'
+              - task: CmdLine@2
+                displayName: Install app hosting dependencies (skip Cypress install)
+                condition: eq(variables.CYPRESS_CACHE_RESTORED, 'true')
+                inputs:
+                  script: |
+                    CYPRESS_INSTALL_BINARY=0 pnpm install --frozen-lockfile
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: Install app hosting dependencies (include Cypress install)
+                condition: eq(variables.CYPRESS_CACHE_RESTORED, 'false')
+                inputs:
+                  script: |
+                    rm -rf $(pnpm store path)
+                    pnpm install --frozen-lockfile
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Build app hosting sdk'
+                inputs:
+                  script: |
+                    pnpm build
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Install client sdk dependencies'
+                inputs:
+                  script: |
+                    pnpm install
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Build client sdk'
+                inputs:
+                  script: |
+                    pnpm build-force-blazor
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Configure host machine'
+                inputs:
+                  script: "sudo chmod -R 755 ./ \npnpm run setup\n"
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Run E2E Perf tests'
+                condition: succeeded()
+                inputs:
+                  targetType: inline
+                  script: 'node tools/cli/runAppsWithE2ETests.js --appUrl=https://localhost:4002 --reportFileName=e2e-tests-report-perf --envType=perf'
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: PublishTestResults@2
+                inputs:
+                  testResultsFormat: 'JUnit'
+                  testResultsFiles: '**/e2e-tests-report*.xml'
+                  testRunTitle: 'E2E Tests - Perf'
+                  mergeTestResults: true
+                condition: succeededOrFailed()
+          - job: E2ETestAndroid
+            displayName: 'E2E Test - Android'
+            pool:
+              name: Azure Pipelines
+              image: internal-macos-11
```

Comment:
Can this be updated to macos-12?

Reply:
I think we can. What do you think? @jekloudaMSFT